src/Middleware/CallableFactory.php

```diff
@@ -46,54 +63,44 @@ public function create(mixed $definition): callable
     }
 
     /**
-     * @param mixed $definition
-     *
-     * @throws ContainerExceptionInterface Error while retrieving the entry from container.
+     * @param string $className
+     * @param string $methodName
      *
      * @return mixed
+     *
+     * @throws ContainerExceptionInterface Error while retrieving the entry from container.
+     * @throws NotFoundExceptionInterface
      */
-    private function prepare(mixed $definition): mixed
+    private function fromDefinition(string $className, string $methodName): mixed
     {
-        if (is_string($definition) && $this->container->has($definition)) {
-            return $this->container->get($definition);
+        if (!class_exists($className) && $this->container->has($className)) {
```

Comment:
Can the first check be omitted and the further fetching from the container (on 97 line) also be removed?